shortening window pane causes app crash and prompts macos to force kill

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Zed Preview app crashes when quickly shortening the window. Causes system to prompt to force kill the app due to large memory allocation.

### Steps to Reproduce

1. Open Zed
2. Lengthen app to take up whole screen
3. Drag window quickly to the left

### Recovery?

I cannot reopen the app (will crash again) since macOS will prompt to report error to Apple. _However_, if i'm quick enough I can lengthen the app window pane wide enough  seconds after re-opening the app to avoid crashing again... 


### Environment

Zed: v0.121.4 (Zed Preview)
OS: macOS 14.3.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature


<img width="433" alt="image" src="https://github.com/zed-industries/zed/assets/21319730/0234ea45-b0ca-4262-badb-50cd9013ba54">

<img width="539" alt="Screenshot 2024-02-06 at 11 30 32 AM" src="https://github.com/zed-industries/zed/assets/21319730/c9844f1e-6de2-4177-98ba-f896c03a38b4">

<img width="958" alt="Screenshot 2024-02-06 at 11 44 24 AM" src="https://github.com/zed-industries/zed/assets/21319730/24d896fe-ea03-483c-8954-98931b9fa907">


### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

<details>

```
2024-02-06T11:36:12-08:00 [INFO] ========== starting zed ==========
2024-02-06T11:36:12-08:00 [INFO] Opening main db
2024-02-06T11:36:12-08:00 [ERROR] crates/theme/src/settings.rs:234: theme not found: Catppuccin Mocha
2024-02-06T11:36:12-08:00 [INFO] Opening main db
2024-02-06T11:36:12-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: expected value at line 1 column 1
2024-02-06T11:36:12-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: Is a directory (os error 21)
2024-02-06T11:36:12-08:00 [INFO] build git repository ".git"
2024-02-06T11:36:12-08:00 [INFO] set status on client 0: Authenticating
2024-02-06T11:36:12-08:00 [INFO] Opening main db
2024-02-06T11:36:12-08:00 [INFO] set status on client 70574: Connecting
2024-02-06T11:36:12-08:00 [INFO] Opening main db
2024-02-06T11:36:12-08:00 [INFO] starting language server "taplo-ls", path: "/Users/andrest.dev/xx/rise/ingestors", id: 1
2024-02-06T11:36:12-08:00 [ERROR] crates/workspace/src/persistence/model.rs:257: failed to read path after update
2024-02-06T11:36:12-08:00 [ERROR] crates/workspace/src/persistence/model.rs:257: failed to read path after update
2024-02-06T11:36:13-08:00 [INFO] set environment variables from shell:/bin/zsh, path:/Applications/Docker.app/Contents/Resources/bin:/Applications/Tailscale.app/Contents/MacOS:/Library/NessusAgent/run/sbin/:/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin:/Users/andrest.dev/.pyenv/plugins/pyenv-virtualenv/shims:/Users/andrest.dev/.pyenv/shims:/Users/andrest.dev/.pyenv/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/andrest.dev/.cargo/bin:/opt/vcpkg
2024-02-06T11:36:13-08:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-02-06T11:36:13-08:00 [INFO] add connection to peer
2024-02-06T11:36:13-08:00 [INFO] waiting for server hello
2024-02-06T11:36:13-08:00 [INFO] got server hello
2024-02-06T11:36:13-08:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 327, id: 1174669 })
2024-02-06T11:36:13-08:00 [INFO] set status on client 70574: Connected { peer_id: PeerId { owner_id: 327, id: 1174669 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-02-06T11:39:11-08:00 [INFO] ========== starting zed ==========
2024-02-06T11:39:11-08:00 [INFO] Opening main db
2024-02-06T11:39:11-08:00 [ERROR] crates/theme/src/settings.rs:234: theme not found: Catppuccin Mocha
2024-02-06T11:39:11-08:00 [INFO] Opening main db
2024-02-06T11:39:11-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: expected value at line 1 column 1
2024-02-06T11:39:11-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: Is a directory (os error 21)
2024-02-06T11:39:11-08:00 [INFO] build git repository ".git"
2024-02-06T11:39:11-08:00 [INFO] set status on client 0: Authenticating
2024-02-06T11:39:11-08:00 [INFO] Opening main db
2024-02-06T11:39:11-08:00 [INFO] set status on client 70574: Connecting
2024-02-06T11:39:11-08:00 [INFO] Opening main db
2024-02-06T11:39:11-08:00 [INFO] starting language server "taplo-ls", path: "/Users/andrest.dev/xx/rise/ingestors", id: 1
2024-02-06T11:39:11-08:00 [INFO] set environment variables from shell:/bin/zsh, path:/Applications/Docker.app/Contents/Resources/bin:/Applications/Tailscale.app/Contents/MacOS:/Library/NessusAgent/run/sbin/:/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin:/Users/andrest.dev/.pyenv/plugins/pyenv-virtualenv/shims:/Users/andrest.dev/.pyenv/shims:/Users/andrest.dev/.pyenv/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/andrest.dev/.cargo/bin:/opt/vcpkg
2024-02-06T11:39:11-08:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-02-06T11:39:12-08:00 [ERROR] {
  "thread": "PTY reader",
  "payload": "index out of bounds: the len is 1 but the index is 1",
  "location_data": {
    "file": "/Users/administrator/.cargo/registry/src/index.crates.io-6f17d22bba15001f/alacritty_terminal-0.21.0/src/grid/mod.rs",
    "line": 431
  },
  "backtrace": [
    "<backtrace::capture::Backtrace>::create",
    "<backtrace::capture::Backtrace>::new",
    "Zed::init_panic_hook::{closure#0}",
    "std::panicking::rust_panic_with_hook",
    "std::panicking::begin_panic_handler::{{closure}}",
    "std::sys_common::backtrace::__rust_end_short_backtrace",
    "_rust_begin_unwind",
    "core::panicking::panic_fmt",
    "core::panicking::panic_bounds_check",
    "<alacritty_terminal::term::Term<terminal::ZedListener> as vte::ansi::Handler>::input",
    "<utf8parse::Parser>::advance::<vte::VtUtf8Receiver<vte::ansi::Performer<alacritty_terminal::term::Term<terminal::ZedListener>, vte::ansi::StdSyncHandler>>>",
    "<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::pty_read::<std::fs::File>",
    "std::sys_common::backtrace::__rust_begin_short_backtrace::<<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::spawn::{closure#0}, (alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>, alacritty_terminal::event_loop::State)>",
    "<<std::thread::Builder>::spawn_unchecked_<<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::spawn::{closure#0}, (alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>, alacritty_terminal::event_loop::State)>::{closure#1} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}",
    "std::sys::unix::thread::Thread::new::thread_start",
    "__pthread_joiner_wake"
  ],
  "app_version": "0.121.4",
  "release_channel": "Zed Preview",
  "os_name": "macOS",
  "os_version": "14.3.0",
  "architecture": "aarch64",
  "panicked_on": 1707248352285,
  "installation_id": "60580ee0-dcd7-4593-9972-3c55b33e720a",
  "session_id": "cfe07e4f-824c-4a8b-9f5c-50136ef6f88c"
}
2024-02-06T11:40:34-08:00 [INFO] ========== starting zed ==========
2024-02-06T11:40:34-08:00 [INFO] Opening main db
2024-02-06T11:40:34-08:00 [ERROR] crates/theme/src/settings.rs:234: theme not found: Catppuccin Mocha
2024-02-06T11:40:34-08:00 [INFO] Opening main db
2024-02-06T11:40:34-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: expected value at line 1 column 1
2024-02-06T11:40:34-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: Is a directory (os error 21)
2024-02-06T11:40:34-08:00 [INFO] set status on client 0: Authenticating
2024-02-06T11:40:34-08:00 [INFO] build git repository ".git"
2024-02-06T11:40:34-08:00 [INFO] set status on client 70574: Connecting
2024-02-06T11:40:34-08:00 [INFO] Opening main db
2024-02-06T11:40:34-08:00 [INFO] Opening main db
2024-02-06T11:40:34-08:00 [INFO] starting language server "taplo-ls", path: "/Users/andrest.dev/xx/rise/ingestors", id: 1
2024-02-06T11:40:34-08:00 [INFO] set environment variables from shell:/bin/zsh, path:/Applications/Docker.app/Contents/Resources/bin:/Applications/Tailscale.app/Contents/MacOS:/Library/NessusAgent/run/sbin/:/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin:/Users/andrest.dev/.pyenv/plugins/pyenv-virtualenv/shims:/Users/andrest.dev/.pyenv/shims:/Users/andrest.dev/.pyenv/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/andrest.dev/.cargo/bin:/opt/vcpkg
2024-02-06T11:40:34-08:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-02-06T11:40:35-08:00 [INFO] add connection to peer
2024-02-06T11:40:35-08:00 [INFO] waiting for server hello
2024-02-06T11:40:35-08:00 [ERROR] {
  "thread": "PTY reader",
  "payload": "index out of bounds: the len is 1 but the index is 1",
  "location_data": {
    "file": "/Users/administrator/.cargo/registry/src/index.crates.io-6f17d22bba15001f/alacritty_terminal-0.21.0/src/grid/mod.rs",
    "line": 431
  },
  "backtrace": [
    "<backtrace::capture::Backtrace>::create",
    "<backtrace::capture::Backtrace>::new",
    "Zed::init_panic_hook::{closure#0}",
    "std::panicking::rust_panic_with_hook",
    "std::panicking::begin_panic_handler::{{closure}}",
    "std::sys_common::backtrace::__rust_end_short_backtrace",
    "_rust_begin_unwind",
    "core::panicking::panic_fmt",
    "core::panicking::panic_bounds_check",
    "<alacritty_terminal::term::Term<terminal::ZedListener> as vte::ansi::Handler>::input",
    "<utf8parse::Parser>::advance::<vte::VtUtf8Receiver<vte::ansi::Performer<alacritty_terminal::term::Term<terminal::ZedListener>, vte::ansi::StdSyncHandler>>>",
    "<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::pty_read::<std::fs::File>",
    "std::sys_common::backtrace::__rust_begin_short_backtrace::<<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::spawn::{closure#0}, (alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>, alacritty_terminal::event_loop::State)>",
    "<<std::thread::Builder>::spawn_unchecked_<<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::spawn::{closure#0}, (alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>, alacritty_terminal::event_loop::State)>::{closure#1} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}",
    "std::sys::unix::thread::Thread::new::thread_start",
    "__pthread_joiner_wake"
  ],
  "app_version": "0.121.4",
  "release_channel": "Zed Preview",
  "os_name": "macOS",
  "os_version": "14.3.0",
  "architecture": "aarch64",
  "panicked_on": 1707248435216,
  "installation_id": "60580ee0-dcd7-4593-9972-3c55b33e720a",
  "session_id": "8234b22d-47a4-4da0-8b88-772becc4066f"
}
2024-02-06T11:40:37-08:00 [INFO] ========== starting zed ==========
2024-02-06T11:40:37-08:00 [INFO] Opening main db
2024-02-06T11:40:37-08:00 [ERROR] crates/theme/src/settings.rs:234: theme not found: Catppuccin Mocha
2024-02-06T11:40:37-08:00 [INFO] Opening main db
2024-02-06T11:40:37-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: expected value at line 1 column 1
2024-02-06T11:40:37-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: Is a directory (os error 21)
2024-02-06T11:40:37-08:00 [INFO] build git repository ".git"
2024-02-06T11:40:37-08:00 [INFO] set status on client 0: Authenticating
2024-02-06T11:40:37-08:00 [INFO] Opening main db
2024-02-06T11:40:37-08:00 [INFO] set status on client 70574: Connecting
2024-02-06T11:40:37-08:00 [INFO] Opening main db
2024-02-06T11:40:37-08:00 [INFO] starting language server "taplo-ls", path: "/Users/andrest.dev/xx/rise/ingestors", id: 1
2024-02-06T11:40:38-08:00 [INFO] set environment variables from shell:/bin/zsh, path:/Applications/Docker.app/Contents/Resources/bin:/Applications/Tailscale.app/Contents/MacOS:/Library/NessusAgent/run/sbin/:/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin:/Users/andrest.dev/.pyenv/plugins/pyenv-virtualenv/shims:/Users/andrest.dev/.pyenv/shims:/Users/andrest.dev/.pyenv/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/andrest.dev/.cargo/bin:/opt/vcpkg
2024-02-06T11:40:38-08:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-02-06T11:40:39-08:00 [INFO] add connection to peer
2024-02-06T11:40:39-08:00 [INFO] waiting for server hello
2024-02-06T11:40:39-08:00 [INFO] got server hello
2024-02-06T11:40:39-08:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 327, id: 1175442 })
2024-02-06T11:40:39-08:00 [INFO] set status on client 70574: Connected { peer_id: PeerId { owner_id: 327, id: 1175442 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-02-06T11:44:14-08:00 [INFO] ========== starting zed ==========
2024-02-06T11:44:14-08:00 [INFO] Opening main db
2024-02-06T11:44:14-08:00 [ERROR] crates/theme/src/settings.rs:234: theme not found: Catppuccin Mocha
2024-02-06T11:44:14-08:00 [INFO] Opening main db
2024-02-06T11:44:14-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: expected value at line 1 column 1
2024-02-06T11:44:14-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: Is a directory (os error 21)
2024-02-06T11:44:14-08:00 [INFO] build git repository ".git"
2024-02-06T11:44:14-08:00 [INFO] set status on client 0: Authenticating
2024-02-06T11:44:14-08:00 [INFO] Opening main db
2024-02-06T11:44:14-08:00 [INFO] set status on client 70574: Connecting
2024-02-06T11:44:14-08:00 [INFO] Opening main db
2024-02-06T11:44:14-08:00 [INFO] starting language server "taplo-ls", path: "/Users/andrest.dev/xx/rise/ingestors", id: 1
2024-02-06T11:44:15-08:00 [INFO] set environment variables from shell:/bin/zsh, path:/Applications/Docker.app/Contents/Resources/bin:/Applications/Tailscale.app/Contents/MacOS:/Library/NessusAgent/run/sbin/:/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin:/Users/andrest.dev/.pyenv/plugins/pyenv-virtualenv/shims:/Users/andrest.dev/.pyenv/shims:/Users/andrest.dev/.pyenv/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/andrest.dev/.cargo/bin:/opt/vcpkg
2024-02-06T11:44:15-08:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-02-06T11:44:15-08:00 [ERROR] {
  "thread": "PTY reader",
  "payload": "index out of bounds: the len is 1 but the index is 1",
  "location_data": {
    "file": "/Users/administrator/.cargo/registry/src/index.crates.io-6f17d22bba15001f/alacritty_terminal-0.21.0/src/grid/mod.rs",
    "line": 431
  },
  "backtrace": [
    "<backtrace::capture::Backtrace>::create",
    "<backtrace::capture::Backtrace>::new",
    "Zed::init_panic_hook::{closure#0}",
    "std::panicking::rust_panic_with_hook",
    "std::panicking::begin_panic_handler::{{closure}}",
    "std::sys_common::backtrace::__rust_end_short_backtrace",
    "_rust_begin_unwind",
    "core::panicking::panic_fmt",
    "core::panicking::panic_bounds_check",
    "<alacritty_terminal::term::Term<terminal::ZedListener> as vte::ansi::Handler>::input",
    "<utf8parse::Parser>::advance::<vte::VtUtf8Receiver<vte::ansi::Performer<alacritty_terminal::term::Term<terminal::ZedListener>, vte::ansi::StdSyncHandler>>>",
    "<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::pty_read::<std::fs::File>",
    "std::sys_common::backtrace::__rust_begin_short_backtrace::<<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::spawn::{closure#0}, (alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>, alacritty_terminal::event_loop::State)>",
    "<<std::thread::Builder>::spawn_unchecked_<<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::spawn::{closure#0}, (alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>, alacritty_terminal::event_loop::State)>::{closure#1} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}",
    "std::sys::unix::thread::Thread::new::thread_start",
    "__pthread_joiner_wake"
  ],
  "app_version": "0.121.4",
  "release_channel": "Zed Preview",
  "os_name": "macOS",
  "os_version": "14.3.0",
  "architecture": "aarch64",
  "panicked_on": 1707248655806,
  "installation_id": "60580ee0-dcd7-4593-9972-3c55b33e720a",
  "session_id": "81d57c8b-4560-4dcf-a03a-3999568e3141"
}
2024-02-06T11:57:13-08:00 [INFO] ========== starting zed ==========
2024-02-06T11:57:13-08:00 [INFO] Opening main db
2024-02-06T11:57:13-08:00 [ERROR] crates/theme/src/settings.rs:234: theme not found: Catppuccin Mocha
2024-02-06T11:57:13-08:00 [INFO] Opening main db
2024-02-06T11:57:13-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: expected value at line 1 column 1
2024-02-06T11:57:13-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: Is a directory (os error 21)
2024-02-06T11:57:13-08:00 [INFO] build git repository ".git"
2024-02-06T11:57:13-08:00 [INFO] set status on client 0: Authenticating
2024-02-06T11:57:13-08:00 [INFO] Opening main db
2024-02-06T11:57:13-08:00 [INFO] set status on client 70574: Connecting
2024-02-06T11:57:13-08:00 [INFO] Opening main db
2024-02-06T11:57:13-08:00 [INFO] starting language server "taplo-ls", path: "/Users/andrest.dev/xx/rise/ingestors", id: 1
2024-02-06T11:57:14-08:00 [INFO] set environment variables from shell:/bin/zsh, path:/Applications/Docker.app/Contents/Resources/bin:/Applications/Tailscale.app/Contents/MacOS:/Library/NessusAgent/run/sbin/:/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin:/Users/andrest.dev/.pyenv/plugins/pyenv-virtualenv/shims:/Users/andrest.dev/.pyenv/shims:/Users/andrest.dev/.pyenv/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/andrest.dev/.cargo/bin:/opt/vcpkg
2024-02-06T11:57:14-08:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-02-06T11:57:14-08:00 [ERROR] {
  "thread": "PTY reader",
  "payload": "index out of bounds: the len is 1 but the index is 1",
  "location_data": {
    "file": "/Users/administrator/.cargo/registry/src/index.crates.io-6f17d22bba15001f/alacritty_terminal-0.21.0/src/grid/mod.rs",
    "line": 431
  },
  "backtrace": [
    "<backtrace::capture::Backtrace>::create",
    "<backtrace::capture::Backtrace>::new",
    "Zed::init_panic_hook::{closure#0}",
    "std::panicking::rust_panic_with_hook",
    "std::panicking::begin_panic_handler::{{closure}}",
    "std::sys_common::backtrace::__rust_end_short_backtrace",
    "_rust_begin_unwind",
    "core::panicking::panic_fmt",
    "core::panicking::panic_bounds_check",
    "<alacritty_terminal::term::Term<terminal::ZedListener> as vte::ansi::Handler>::input",
    "<utf8parse::Parser>::advance::<vte::VtUtf8Receiver<vte::ansi::Performer<alacritty_terminal::term::Term<terminal::ZedListener>, vte::ansi::StdSyncHandler>>>",
    "<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::pty_read::<std::fs::File>",
    "std::sys_common::backtrace::__rust_begin_short_backtrace::<<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::spawn::{closure#0}, (alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>, alacritty_terminal::event_loop::State)>",
    "<<std::thread::Builder>::spawn_unchecked_<<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::spawn::{closure#0}, (alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>, alacritty_terminal::event_loop::State)>::{closure#1} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}",
    "std::sys::unix::thread::Thread::new::thread_start",
    "__pthread_joiner_wake"
  ],
  "app_version": "0.121.4",
  "release_channel": "Zed Preview",
  "os_name": "macOS",
  "os_version": "14.3.0",
  "architecture": "aarch64",
  "panicked_on": 1707249434605,
  "installation_id": "60580ee0-dcd7-4593-9972-3c55b33e720a",
  "session_id": "59ae2b07-4a87-4062-97af-0a72d77e76fb"
}
2024-02-06T11:57:16-08:00 [INFO] ========== starting zed ==========
2024-02-06T11:57:16-08:00 [INFO] Opening main db
2024-02-06T11:57:17-08:00 [ERROR] crates/theme/src/settings.rs:234: theme not found: Catppuccin Mocha
2024-02-06T11:57:17-08:00 [INFO] Opening main db
2024-02-06T11:57:17-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: expected value at line 1 column 1
2024-02-06T11:57:17-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: Is a directory (os error 21)
2024-02-06T11:57:17-08:00 [INFO] set status on client 0: Authenticating
2024-02-06T11:57:17-08:00 [INFO] build git repository ".git"
2024-02-06T11:57:17-08:00 [INFO] set status on client 70574: Connecting
2024-02-06T11:57:17-08:00 [INFO] Opening main db
2024-02-06T11:57:17-08:00 [INFO] Opening main db
2024-02-06T11:57:17-08:00 [INFO] starting language server "taplo-ls", path: "/Users/andrest.dev/xx/rise/ingestors", id: 1
2024-02-06T11:57:17-08:00 [INFO] set environment variables from shell:/bin/zsh, path:/Applications/Docker.app/Contents/Resources/bin:/Applications/Tailscale.app/Contents/MacOS:/Library/NessusAgent/run/sbin/:/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin:/Users/andrest.dev/.pyenv/plugins/pyenv-virtualenv/shims:/Users/andrest.dev/.pyenv/shims:/Users/andrest.dev/.pyenv/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/andrest.dev/.cargo/bin:/opt/vcpkg
2024-02-06T11:57:17-08:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-02-06T11:57:18-08:00 [ERROR] {
  "thread": "PTY reader",
  "payload": "index out of bounds: the len is 1 but the index is 1",
  "location_data": {
    "file": "/Users/administrator/.cargo/registry/src/index.crates.io-6f17d22bba15001f/alacritty_terminal-0.21.0/src/grid/mod.rs",
    "line": 431
  },
  "backtrace": [
    "<backtrace::capture::Backtrace>::create",
    "<backtrace::capture::Backtrace>::new",
    "Zed::init_panic_hook::{closure#0}",
    "std::panicking::rust_panic_with_hook",
    "std::panicking::begin_panic_handler::{{closure}}",
    "std::sys_common::backtrace::__rust_end_short_backtrace",
    "_rust_begin_unwind",
    "core::panicking::panic_fmt",
    "core::panicking::panic_bounds_check",
    "<alacritty_terminal::term::Term<terminal::ZedListener> as vte::ansi::Handler>::input",
    "<utf8parse::Parser>::advance::<vte::VtUtf8Receiver<vte::ansi::Performer<alacritty_terminal::term::Term<terminal::ZedListener>, vte::ansi::StdSyncHandler>>>",
    "<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::pty_read::<std::fs::File>",
    "std::sys_common::backtrace::__rust_begin_short_backtrace::<<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::spawn::{closure#0}, (alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>, alacritty_terminal::event_loop::State)>",
    "<<std::thread::Builder>::spawn_unchecked_<<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::spawn::{closure#0}, (alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>, alacritty_terminal::event_loop::State)>::{closure#1} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}",
    "std::sys::unix::thread::Thread::new::thread_start",
    "__pthread_joiner_wake"
  ],
  "app_version": "0.121.4",
  "release_channel": "Zed Preview",
  "os_name": "macOS",
  "os_version": "14.3.0",
  "architecture": "aarch64",
  "panicked_on": 1707249438016,
  "installation_id": "60580ee0-dcd7-4593-9972-3c55b33e720a",
  "session_id": "81dc0a76-b64b-4b8e-8acd-9fbcd49b3cb2"
}
2024-02-06T11:57:21-08:00 [INFO] ========== starting zed ==========
2024-02-06T11:57:21-08:00 [INFO] Opening main db
2024-02-06T11:57:21-08:00 [ERROR] crates/theme/src/settings.rs:234: theme not found: Catppuccin Mocha
2024-02-06T11:57:21-08:00 [INFO] Opening main db
2024-02-06T11:57:21-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: expected value at line 1 column 1
2024-02-06T11:57:21-08:00 [ERROR] /Users/administrator/actions-runner-2/_work/zed/zed/crates/theme/src/registry.rs:260: Is a directory (os error 21)
2024-02-06T11:57:21-08:00 [INFO] set status on client 0: Authenticating
2024-02-06T11:57:21-08:00 [INFO] build git repository ".git"
2024-02-06T11:57:21-08:00 [INFO] set status on client 70574: Connecting
2024-02-06T11:57:21-08:00 [INFO] Opening main db
2024-02-06T11:57:21-08:00 [INFO] Opening main db
2024-02-06T11:57:21-08:00 [INFO] starting language server "taplo-ls", path: "/Users/andrest.dev/xx/rise/ingestors", id: 1
2024-02-06T11:57:21-08:00 [INFO] set environment variables from shell:/bin/zsh, path:/Applications/Docker.app/Contents/Resources/bin:/Applications/Tailscale.app/Contents/MacOS:/Library/NessusAgent/run/sbin/:/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin:/Users/andrest.dev/.pyenv/plugins/pyenv-virtualenv/shims:/Users/andrest.dev/.pyenv/shims:/Users/andrest.dev/.pyenv/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/andrest.dev/.cargo/bin:/opt/vcpkg
2024-02-06T11:57:22-08:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-02-06T11:57:22-08:00 [ERROR] {
  "thread": "PTY reader",
  "payload": "index out of bounds: the len is 1 but the index is 1",
  "location_data": {
    "file": "/Users/administrator/.cargo/registry/src/index.crates.io-6f17d22bba15001f/alacritty_terminal-0.21.0/src/grid/mod.rs",
    "line": 431
  },
  "backtrace": [
    "<backtrace::capture::Backtrace>::create",
    "<backtrace::capture::Backtrace>::new",
    "Zed::init_panic_hook::{closure#0}",
    "std::panicking::rust_panic_with_hook",
    "std::panicking::begin_panic_handler::{{closure}}",
    "std::sys_common::backtrace::__rust_end_short_backtrace",
    "_rust_begin_unwind",
    "core::panicking::panic_fmt",
    "core::panicking::panic_bounds_check",
    "<alacritty_terminal::term::Term<terminal::ZedListener> as vte::ansi::Handler>::input",
    "<utf8parse::Parser>::advance::<vte::VtUtf8Receiver<vte::ansi::Performer<alacritty_terminal::term::Term<terminal::ZedListener>, vte::ansi::StdSyncHandler>>>",
    "<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::pty_read::<std::fs::File>",
    "std::sys_common::backtrace::__rust_begin_short_backtrace::<<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::spawn::{closure#0}, (alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>, alacritty_terminal::event_loop::State)>",
    "<<std::thread::Builder>::spawn_unchecked_<<alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>>::spawn::{closure#0}, (alacritty_terminal::event_loop::EventLoop<alacritty_terminal::tty::unix::Pty, terminal::ZedListener>, alacritty_terminal::event_loop::State)>::{closure#1} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}",
    "std::sys::unix::thread::Thread::new::thread_start",
    "__pthread_joiner_wake"
  ],
  "app_version": "0.121.4",
  "release_channel": "Zed Preview",
  "os_name": "macOS",
  "os_version": "14.3.0",
  "architecture": "aarch64",
  "panicked_on": 1707249442471,
  "installation_id": "60580ee0-dcd7-4593-9972-3c55b33e720a",
  "session_id": "edc97e85-54bd-4e23-a5c4-939b91d3e409"
}
```
</details>

--- a/crates/terminal_view/src/terminal_element.rs
+++ b/crates/terminal_view/src/terminal_element.rs
@@ -450,6 +450,13 @@ impl TerminalElement {
             let mut size = bounds.size.clone();
             size.width -= gutter;
 
+            // https://github.com/zed-industries/zed/issues/2750
+            // if the terminal is one column wide, rendering 🦀
+            // causes alacritty to misbehave.
+            if size.width < cell_width * 2.0 {
+                size.width = cell_width * 2.0;
+            }
+
             TerminalSize::new(line_height, cell_width, size)
         };
 

